internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -501,23 +503,33 @@ func (r *GitJobReconciler) deleteJobIfNeeded(ctx context.Context, gitRepo *v1alp
 	// the following cases imply that the job is still running but we need to stop it and
 	// create a new one
 	if gitRepo.Spec.ForceSyncGeneration != gitRepo.Status.UpdateGeneration {
-		jobDeletedMessage := "job deletion triggered because of ForceUpdateGeneration"
-		logger.Info(jobDeletedMessage)
-		if err := r.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil && !errors.IsNotFound(err) {
-			return err, true
+		if forceSync, ok := job.Labels[forceSyncGenerationLabel]; ok {
+			strForceSync := fmt.Sprintf("%d", gitRepo.Spec.ForceSyncGeneration)
+			if strForceSync != forceSync {
+				jobDeletedMessage := "job deletion triggered because of ForceUpdateGeneration"
+				logger.Info(jobDeletedMessage)
```

Comment:
I would log this in debug mode only, same for the other "job deleted" messages:
```suggestion
				logger.V(1).Info(jobDeletedMessage)
```